packages/workflow/src/interfaces.ts

```diff
@@ -872,11 +872,17 @@ export type CronContext = {
 	nodeId: string;
 	workflowId: string;
 	timezone: string;
-	expression: CronExpression;
+	expression: CronExpression | ITimeSource;
 	recurrence?: CronRecurrenceRule;
 };
 
-export type Cron = { expression: CronExpression; recurrence?: CronRecurrenceRule };
+export interface ITimeSource {
+	getTimeSource(): string | Date;
+	updateTimeSource(): void;
+	toString(): string;
+}
+
+export type Cron = { expression: CronExpression | ITimeSource; recurrence?: CronRecurrenceRule };
```

Comment:
Is `expression` still a fitting name if one of the union constituents is a class? Should the string expression be a time source itself so callers don't have to be checking what this is?